File: study/1주차 문제/이다영/문자열 내 마음대로 정렬하기.java
package fin;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;
import java.util.List;

// 문자열 내 마음대로 정렬하기
public class Pro_12915 {

	public static void main(String[] args) {
		solution(new String[]{"sun", "bed", "car"}, 1);
		solution(new String[]{"abce", "abcd", "cdx"}, 2);
	}
	
	public static String[] solution(String[] strings, int n) {
		List<String> stringList = new ArrayList<>(Arrays.asList(strings));
		
		Collections.sort(stringList);
		Collections.sort(stringList, (o1, o2) -> o1.charAt(n) - o2.charAt(n));
		
        return stringList.toArray(new String[stringList.size()]);
    }
}

File: study/1주차 문제/박정환/문자열내마음대로정렬하기.py
def solution(strings, n):
    answer = []  
    strings.sort()               
    a = []          
    
    for i in strings:
        a.append(i[n])
    a.sort()        
    
    for i in a:
        for j in strings:
            if i == j[n]:    
                answer.append(j)
                strings.remove(j)
                break              
            
    return answer
